### ISSUE TITLE
Feature request: Prevent screenshotting inside app

### ISSUE THREAD

**Issue by benefaktorix** [guardianproject/ChatSecureAndroid#704]
Textsecure has some nice features like preventing making screenshots inside the app and blocking viewing the content of messages in the "recent apps" list. Would be great to have it here also!

**Comment by Manouchehri** [comment]
The idea that a screenshot can be considered "proof" is ridiculous to begin with, I definitely don't want ChatSecure to reaffirm that myth.

I know TextSecure did it because they wanted to give users an option to avoid accidentally saving a log to an uncontrolled local storage. Still, I think this does more harm than good.

If somebody still wants to add it, it's a [few lines to set](https://github.com/WhisperSystems/TextSecure/blob/bb5fe39d3dc928a8a1e0f9a8c8f5deec9b7ccf32/src/org/thoughtcrime/securesms/PassphraseRequiredActionBarActivity.java#L184).

**Comment by n8fr8** [comment]
Yes, we are adding this and other features (like hiding from your recent apps, and changing the icon in the launcher) as part of a new set of physical security features in our v15 work.

**Comment by chseluv** [comment]
Manoucheri, it is not ridiculous at all. Inside a screenshot there can be enough proof in critical conditions to jeopardize users' security in case sensitive data is being transmitted. This is just common sense. It would be good to enable screenshots but let the recipient give the requesting user the permission to actually take it or at least the recipient should be warned about the status of screenshot activation in sender's client.

**Comment by Manouchehri** [comment]
If an application denies you from taking a screenshot, you can simply use another phone to take a picture of the screen. There is no logical way of preventing a remote recipient from sharing a message.

As far as I'm aware, there's no plaintext attack against OTR that would risk future conversations. You can freely share old conversations without worrying about your keys.

Again, by reaffirming the thought that a screenshot is proof, we're collectively pretending that OTR deniability doesn't exist.

https://whispersystems.org/blog/simplifying-otr-deniability/
